src/query/shorthand.ts

```diff
@@ -121,6 +121,13 @@ export function fieldDef(encQ: EncodingQuery,
         value: value(encQ.bin['maxbins'], replace[Property.BIN_MAXBINS])
       });
     }
+  } else if (include[Property.SORT] && encQ.sort && !isEnumSpec(encQ.sort)) {
+    if (!(encQ.sort as SortFieldQuery)['op']) {
```

Comment:
I'm also not sure why we need this branch for sort but not for scale?  What's different?  (I mean you might be right, but can you explain?)